src/libopensc/opensc.h

```diff
@@ -940,9 +946,11 @@ int sc_cancel(sc_context_t *ctx);
 /**
  * Tries acquire the reader lock.
  * @param  card  The card to lock
+ * @param  resetok If nonzero means to continue even if the card was reset
+ *	as long as it wasn't reset between individual operations
  * @retval SC_SUCCESS on success
  */
-int sc_lock(struct sc_card *card);
+int sc_lock(struct sc_card *card, int resetok);
```

Comment:
I agree with @frankmorgner  on the card_reader_lock_obtained. 
https://github.com/OpenSC/OpenSC/commit/84a69ce2ba100be8121218a61a5d549ea3cbdbfb introduced the code. It puts the card driver in control as each card has its own quirks.  You need to write a card_reader_lock_obtained  for your driver to do what you need to do when it has been reset. The point of letting the card driver know immediately, it can take action before any other PC/SC command or APDU is issued and attempt to recover the state.   

The card-piv.c is the only card that currently implements a card_reader_lock_obtained. 

"An alternative approach would be to set a flag in the card if a reset has happened." That could be a default card_reader_lock_obtained routine. But then it is still up to the driver to do something with the flag.